### PR TITLE
Add ability to make OP_RETURN commitments

### PR DESCRIPTION
**Pull request by benthecarman**
The OP_RETURN commitment should probably be given a proper `ScriptPubKey` type, but was just hacking this together for funzies.

Proof it works as well: 
https://blockstream.info/testnet/tx/7080d81c2943167380337689664a325ded8ae4ff62c377f18da976fb3516e421?expand

### CODE REVIEW

File: app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala [hunk 1]
```diff
@@ -505,6 +543,11 @@ object CliCommand {
       amount: Bitcoins,
       satoshisPerVirtualByte: Option[SatoshisPerVirtualByte])
       extends CliCommand
+  case class OpReturnCommit(
+      message: String,
+      hashMessage: Boolean,
+      satoshisPerVirtualByte: Option[SatoshisPerVirtualByte])
```
Comment (nkohen) [inline app/cli/src/main/scala/org/bitcoins/cli/ConsoleCli.scala]: nit: should probably be `feeOpt`

File: app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala [hunk 2]
```diff
@@ -477,6 +477,29 @@ class RoutesSpec
 
     }
 
+    "make an OP_RETURN commitment" in {
+
+      val message = "Never gonna give you up, never gonna let you down"
```
Comment (nkohen) [inline app/server-test/src/test/scala/org/bitcoins/server/RoutesSpec.scala:708]: nice :)

File: wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala [hunk 3]
```diff
@@ -262,6 +266,31 @@ abstract class Wallet
...
+      CryptoUtil.sha256(ByteVector(message.getBytes)).bytes
+    } else {
+      if (message.length > 32) {
+        throw new IllegalArgumentException(
+          s"Message cannot be greater than 32 characters, it should be hashed, got $message")
+      } else ByteVector(message.getBytes)
+    }
+
+    val asm = Seq(OP_RETURN,
+                  BytesToPushOntoStack(messageToUse.size),
```
Comment (nkohen) [inline wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: this won't work for small things (think 1 bit), you should use the calculate push op method in script util

File: wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala [hunk 4]
```diff
@@ -262,6 +266,31 @@ abstract class Wallet
...
+      message: String,
+      hashMessage: Boolean,
+      feeRate: FeeUnit,
+      fromAccount: AccountDb): Future[Transaction] = {
+    val messageToUse = if (hashMessage) {
+      CryptoUtil.sha256(ByteVector(message.getBytes)).bytes
+    } else {
+      if (message.length > 32) {
+        throw new IllegalArgumentException(
+          s"Message cannot be greater than 32 characters, it should be hashed, got $message")
```
Comment (nkohen) [inline wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: Are you not allowed to push larger messages?

Comment (benthecarman) [reply wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: I believe 40 bytes is the max, figured 32 bytes is fine for this so we have a clear distinction between hashes.

If you want to push bigger messages checkout BSV :P

Comment (benthecarman) [reply wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: Actually looks like it's 83 bytes

https://github.com/bitcoin/bitcoin/blob/master/src/script/standard.h#L33

Comment (benthecarman) [reply wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala]: I agree then, this should be changed

**Pull request merged**